src/TupleTypes.jl

```diff
@@ -2,15 +2,18 @@ module TupleTypes
 
 import Base: isvatuple
 
-function parameters{T<:Tuple}(::Type{T})
-    if isvatuple(T) || T===Tuple || T===NTuple
-        throw(ArgumentError("elements of $T are undefined"))
-    end
-    (T.parameters...)
+
+check{T}(::Type{T}) = (T===Tuple || T===NTuple) && throw(ArgumentError("parameters of $T are undefined"))
```

Comment:
Yep, this sent me down a rabbit hole: https://github.com/JuliaLang/julia/issues/11715